base/compiler/ssair/passes.jl

```diff
@@ -1148,8 +1148,8 @@ function try_resolve_finalizer!(ir::IRCode, idx::Int, finalizer_idx::Int, defuse
     function note_block_use!(usebb::Int, useidx::Int)
         new_bb_insert_block = nearest_common_dominator(get!(lazypostdomtree),
             bb_insert_block, usebb)
-        if new_bb_insert_block == usebb
-            if usebb == bb_insert_block
+        if new_bb_insert_block == bb_insert_block == usebb
```

Comment:
I think we can optimize this code slightly by avoiding computing `nearest_common_dominator` for the `bb_insert_block == usebb` case? As far as I understand, `bb_insert_block` stays same for that case (assuming a basic block [post-]dominates itself always.

Reply:
Yes, but the nearest common dominator will be trivial to compute in that case, so I'd rather avoid introducing and extra special case here.

Reply:
If you really want, you can add a fast path to nearest_common_dominator